src/program/latex/mod.rs

```diff
@@ -0,0 +1,1600 @@
+                    // relationships require at least two qubits
+                    if relationship.len() < 2 {
+                        continue 'column;
+                    }
+
+                    // the last qubit is the targ
+                    if *qubit == relationship[relationship.len() - 1] {
+                        if let Some(wire) = self.circuit.get_mut(qubit) {
+                            // insert as target at this column
+                            wire.targ.insert(c, true);
+
+                            // set 'column loop targ variable to this targ that
+                            // control qubits will find distance from on their
+                            // respective wires
+                            targ = Some(wire.name)
+                        }
+                    // all other qubits are the controls
+                    } else {
+                        if let Some(wire) = self.circuit.get_mut(qubit) {
+                            // insert as control at this column with initial
+                            // value 0, targeting themselves
+                            wire.ctrl.insert(c, 0);
+
+                            // push ctrl to 'column loop ctrl variables with
+                            // initial value requiring update based on targ
+                            ctrls.push(wire.name);
+                        }
+                    }
+                }
+            } else {
+                // no relationships found on this column, go to next
+                continue 'column;
+            }
+
+            // determine the physical vector where a positive vector points
+            // from control to target, negative, from target to control. The
+            // magnitude of the vector is the absolute value of the distance
+            // between them
+            if let Some(targ) = targ {
+                // distance between qubits is the space between the ctrl and
+                // targ qubits in the circuit
+                for ctrl in ctrls {
+                    // represent inclusive [open, close] brackets of a range
+                    let mut open = None; // opening qubit in range
+                    let mut close = None; // closing qubit in range
+
+                    // find the range between the qubits
+                    let mut i = 0;
+                    for wire in &self.circuit {
+                        // get each existing qubit in the circuit
+                        if *wire.0 == ctrl || *wire.0 == targ {
+                            // if the qubit is the ctrl or target
+                            if let Some(_) = open {
+                                close = Some(i);
+                                break;
+
+                            // open qubit in range not found, set open qubit
+                            } else {
+                                open = Some(i)
+                            }
+                        }
+
+                        i += 1;
```

Comment:
Two things:

- `Iterator::enumerate()` will return an `i` for you so you don't need to calculate it yourself
- Prefer `Option::is_some()` if you don't care about the contained value

```suggestion
                    for (i, wire) in self.circuit.iter().enumerate() {
                        // get each existing qubit in the circuit
                        if *wire.0 == ctrl || *wire.0 == targ {
                            // if the qubit is the ctrl or target
                            if open.is_some() {
                                close = Some(i);
                                break;

                            // open qubit in range not found, set open qubit
                            } else {
                                open = Some(i)
                            }
                        }
```